### PR TITLE
feat: skip governance checks for blocks below the best chainlock

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
A node can miss governance trigger sometimes and then it would stuck not being able to sync any further. This issue can be fixed manually by resetting sync status and reconsidering the "invalid" block. However, that's inconvenient. Also, what it does under the hood is it simply disables some parts of block validation. We could do that automagically and more precise if we would trust ChainLocks instead.

## What was done?
Skip governance checks for blocks below the best known chainlock, add tests.

## How Has This Been Tested?
Run tests.

## Breaking Changes
n/a

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by PastaPastaPasta — DISMISSED** [review]
Concept ACK and utACK 3c3489d7a1d3ac67f40ba262af070d69e012d611

File: test/functional/feature_governance_cl.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018-2024 The Dash Core developers
```
Comment (knst) [inline test/functional/feature_governance_cl.py:2]: nit: copyright 2024

**Review by knst — COMMENTED** [review]
that's strange, I tried to revert this commit, but new functional test `feature_governance_cl.py` still succeed:

```
commit 41ab95dbf8baf7bc0ba4784acc5e89eeae5d9045
Author: UdjinM6 <UdjinM6@users.noreply.github.com>
Date:   Fri May 24 00:52:32 2024 +0300

    feat: skip governance checks for blocks below the best chainlock
```

**Review by thephez — COMMENTED** [review]
Concept ACK 👍

File: src/validation.cpp [hunk 2]
```diff
@@ -2445,7 +2445,9 @@ bool CChainState::ConnectBlock(const CBlock& block, BlockValidationState& state,
     int64_t nTime5_3 = GetTimeMicros(); nTimeCreditPool += nTime5_3 - nTime5_2;
     LogPrint(BCLog::BENCHMARK, "      - CheckCreditPoolDiffForBlock: %.2fms [%.2fs (%.2fms/blk)]\n", MILLI * (nTime5_3 - nTime5_2), nTimeCreditPool * MICRO, nTimeCreditPool * MILLI / nBlocksTotal);
 
-    if (!m_chain_helper->mn_payments->IsBlockValueValid(block, pindex->nHeight, blockSubsidy + feeReward, strError)) {
+    const bool check_superblock = m_clhandler->GetBestChainLock().getHeight() < pindex->nHeight;
+
+    if (!m_chain_helper->mn_payments->IsBlockValueValid(block, pindex->nHeight, blockSubsidy + feeReward, strError, check_superblock)) {
```
Comment (PastaPastaPasta) [inline src/validation.cpp:2450]: how about
```suggestion
    if (!m_chain_helper->mn_payments->IsBlockValueValid(block, pindex->nHeight, blockSubsidy + feeReward, strError, /*check_superblock=*/ m_clhandler->GetBestChainLock().getHeight() < pindex->nHeight)) {
```

Comment (UdjinM6) [reply src/validation.cpp:2450]: we use it twice so I think I'd prefer to avoid code duplication and potential race conditions

**Review by knst — APPROVED** [review]
utACK 18328279ec20b8090a13f5948d1a6ed839c1dae3

**Review by PastaPastaPasta — APPROVED** [review]
utACK 18328279ec20b8090a13f5948d1a6ed839c1dae3

**Pull request merged**